### DISCUSSION TITLE
GitHub App Manifest Flow for single file read/write permission

### DISCUSSION THREAD

**Discussion by stevepentland** [community/community#24586] [Apps, API and Webhooks]
<p>When creating a GitHub App using the manifest flow is there any way to specify the permission level (read/write/none) and path for <code>single_file</code> permissions?</p>
<p>I am currently investigating using the probot base template, which results in an app.yml manifest. However there is no entry for <code>single_file</code> in there. If I add </p>
<pre><code>single_file: write
</code></pre>
<p>or </p>
<pre><code>single_file: read
</code></pre>
<p>I get an error stating that name path is requried</p>
<p><img src="https://user-images.githubusercontent.com/8147610/181085338-148bc1e5-fc5f-4713-a72f-b73e752eb7a2.png" alt="manifest-error.png" data-base62-sha1="3jRIowLhdC3rhpP8Utje8nmG8Fe" width="511" height="97"></p>
<p>Is there any way to include this in the manifest format?</p>

**Comment by stevepentland** [comment, marked as answer]
<p>After analyzing the update behavior for an existing test app, I noted that this can be achieved by:</p>
<pre><code>default_permissions:
  # other permissions...
  single_file: write
</code></pre>
<p>and then in the standard metadata for the manifest (where name, url, description, etc. are):</p>
<pre><code>singe_file_name: &lt;path to the file&gt;
</code></pre>
<p>and this gave the expected behavior via the manifest flow.</p>

**Comment by king468** [comment]
<p>Github app manifest flow for single fille read/</p>
